instance/gandi.py

```diff
@@ -100,9 +101,17 @@ def set_dns_record(self, **record):
             record['ttl'] = 1200
 
         with cache.lock('gandi_set_dns_record'): # Only do one DNS update at a time
-            logger.info('Setting DNS record: %s', record)
-            new_zone_version = self.create_new_zone_version()
-            self.delete_dns_record(new_zone_version, record['name'])
-            returned_record = self.add_dns_record(new_zone_version, record)
-            self.set_zone_version(new_zone_version)
+            for i in range(1, retries + 1):
+                try:
+                    logger.info('Setting DNS record: %s (attempt %d out of %d)', record, i, retries)
+                    new_zone_version = self.create_new_zone_version()
+                    self.delete_dns_record(new_zone_version, record['name'])
+                    returned_record = self.add_dns_record(new_zone_version, record)
+                    self.set_zone_version(new_zone_version)
+                    break
+                except xmlrpc.client.Fault:
+                    time.sleep(retry_delay)
+                    retry_delay *= 2
+                    if i == retries:
```

Comment:
@smarnach might make sense to check and raise before sleeping - this way it would fail earlier.